Documentation/network/egress-gateway/egress-gateway-troubleshooting.rst

```diff
@@ -0,0 +1,60 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _egress_gateway_troubeshooting:
+
+Egress Gateway Advanced Troubleshooting
+=======================================
+
+This document explains various issues users may encounter using egress-gateway.
+
+
+.. _snat_connection_limits:
+
+SNAT Connection Limits
+----------------------
+
+In use-cases where egress-gateway is being used to masquerade traffic to a small set of remote endpoints, it's possible
+to cause issues by exceeding the max number of source IPs that can be allocated by Cilium's NAT mapping per remote endpoint.
```

Comment:
is it possible add a little explanation why this happens (specifically why small set of remote endpoints)?

Reply:
nvm, should've read a bit more before commenting